src/test/ui/consts/control-flow/drop.rs

```diff
@@ -0,0 +1,43 @@
+// Ensure that a value is never dropped inside a `const`.
+
+#![feature(const_if_match)]
+
+// OK: `x` is always moved into the final value and is not dropped inside the initializer.
```

Comment:
I would prefer if we could separate the good/bad parts of the test into different files (general note).

Reply:
It's now done for this test and I'll do this in the future.